MemoryFileWriteStream fix

kamk opened the pull request:
In some methods `stream` instance variable was used directly. This should resolve #2335 issue.

kamk committed:
MemoryFileWriteStream fix
Always using "stream" accessor instead of  instance variable.

guillep commented:
Hi @kamk Thanks! this is great!

Your analysis seems correct. I see other accesses to `stream` in that file are through the accessor, which implements a lazy initialization. The code looks ok.

Would you mind adding a test too? I thin in issue #2335 there are a couple of snippets of code that should fail without this fix. So it should be enough to design a (or many?) test(s) based on those snippets.

Thanks again!

kamk committed:
MemoryFileWriteStream - fix #2
- removed readInto:startingAt:count as this is write stream method
- added test for writing and reading back the data

kamk commented:
Also see PR #4601.

I guess this class could be inherited from `WriteStream` and have implemented methods related to `file` instance variable.

Ducasse commented:
Thanks this is super nice. 
I have a stupid question just reading late the code in github. 

why the following code 
```
^ stream readInto: aCollection startingAt: anInteger count: anInteger3 
 => self stream position: anInteger
```
is not 
```
^ stream readInto: aCollection startingAt: anInteger count: anInteger3 
=>  self stream readInto: aCollection startingAt: anInteger count: anInteger3 
```

Ducasse commented:
Another question just browsing the code is why?

```
position: anInteger
	"Set the current position for accessing the objects to be anInteger, as long 
	as anInteger is within the bounds of the receiver's contents. If it is not, 
	create an error notification."

	stream position: anInteger
=> self stream position: anInteger
```

guillep reviewed (approved):
I think this is a good enough change. Further improvements can be made later (and now we have more tests :))
